tests/components/smlight/test_coordinator.py

```diff
@@ -0,0 +1,59 @@
+"""Tests for SMLIGHT coordinator."""
+
+import socket
+from unittest.mock import MagicMock
+
+from pysmlight.exceptions import SmlightConnectionError
+import pytest
+
+from homeassistant.core import HomeAssistant
+from homeassistant.util.network import is_ip_address
+
+from .conftest import MOCK_HOST
+
+from tests.common import MockConfigEntry
+
+pytestmark = [
+    pytest.mark.usefixtures(
+        "setup_platform",
+        "mock_smlight_client",
+    )
+]
+
+
+async def test_coordinator_get_hostname(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Test coordinator get hostname from IP."""
+    coordinator = mock_config_entry.runtime_data
```

Comment:
@joostlek I not entirely clear what to do here instead? Should I create a new instance of the Coordinator just for the test? Or perhaps test these at a higher level as part of config_flow or async_setup?

Reply:
Neither. The coordinator is just a tool to receive data, we consider that as internals so I would strongly discourage not to use a test_coordinator.py. It all depends on what you're testing, because it will fetch data, and it will display state using that data. So with what you're testing here, where does the user notice this?

Reply:
ok got it. Come to think of it, I can probably drop the hostname handling entirely, it was only used for a name in the device registry. I will also remove test_coordinator.